parlai/crowdsourcing/utils/tests.py

```diff
@@ -308,61 +344,38 @@ def _test_agent_states(
                     'episode_done': False,
                 },
             )
+            self.await_channel_requests()
 
         # Submit the HIT
         for agent_id in agent_ids:
-            self.server.send_agent_act(
-                agent_id=agent_id,
-                act_content={
-                    'task_data': {'final_data': {}},
-                    'MEPHISTO_is_submit': True,
-                },
-            )
+            self.server.submit_mock_unit(agent_id, {'final_data': {}})
+            self.await_channel_requests()
 
         # # Check that the inputs and outputs are as expected
 
-        # Wait until all messages have arrived
-        wait_time = 5.0  # In seconds
-        max_num_tries = 30  # max_num_tries * wait_time is the max time to wait
-        num_tries = 0
-        while num_tries < max_num_tries:
-            actual_states = [agent.state.get_data() for agent in self.db.find_agents()]
-            assert len(actual_states) == len(expected_states)
-            expected_num_messages = sum(
-                len(state['outputs']['messages']) for state in expected_states
-            )
-            actual_num_messages = sum(
-                len(state['outputs']['messages']) for state in actual_states
-            )
-            if expected_num_messages == actual_num_messages:
-                break
-            else:
-                num_tries += 1
-                print(
-                    f'The expected number of messages is '
-                    f'{expected_num_messages:d}, but the actual number of messages '
-                    f'is {actual_num_messages:d}! Waiting for {wait_time:0.1f} seconds '
-                    f'for more messages to arrive (try #{num_tries:d} of '
-                    f'{max_num_tries:d})...'
-                )
-                time.sleep(wait_time)
-        else:
-            actual_num_messages = sum(
-                len(state['outputs']['messages']) for state in actual_states
-            )
-            print(f'\nPrinting all {actual_num_messages:d} messages received:')
-            for state in actual_states:
-                for message in state['outputs']['messages']:
-                    print(message)
+        # Get and filter actual messages
+        time.sleep(self.message_sleep_time)
```

Comment:
@JackUrb Adding this in because, for the model image chat task, the messages only arrive in the agent state a few seconds after the calls to `self.await_channel_requests()`. This is probably because one of the messages contains a large encoded image file